src/semanticlayertools/linkage/cocitation.py

```diff
@@ -0,0 +1,171 @@
+import os
+import time
+import re
+import multiprocessing
+from itertools import combinations
+from collections import Counter
+from typing import TypeVar
+
+import igraph as ig
+import pandas as pd
+import numpy as np
+from tqdm import tqdm
+
+num_processes = multiprocessing.cpu_count()
+
+limitRefLength = TypeVar('limitRefLength', bool, int)
+debugVar = TypeVar('debugVar', bool, str)
+
+
+class Cocitations():
+    """Create cocitation networks.
+
+    Calculates all combinations of all references of publications in given
+    corpus file(s). Can be limited for maximal number of references to consider
+    (e.g. papers with less then 200 references), to speed up creation of
+    networks.
+
+    For each corpus file, graphs are generated by the weighted cocitation tuples,
+    using the Igraph package. Information on obtained clusters are written to
+    '_graphMetadata.txt' files. The subgraph of the Giant component is saved in
+    Pajek format with the ending '_GC.net'. The full edge data is written in
+    edge-Format to a '.ncol' file.
+
+    :param inpath: Path for input data
+    :type inpath: str
+    :param outpath: Path for writing output data
+    :type outpath: str
+    :param columnName: Column name containing the references of a publication
+    :type columnName: str
+    :param numberProc: Number of CPUs the package is allowed to use (default=all)
+    :type numberProc: int
+    :param limitRefLength: Either False or integer giving the maximum number of references a considered publication is allowed to contain
+    :type limitRefLength: bool or int
+    :param timerange: Time range to consider (default=(1945,2005))
+    :type timerange: tuple
+    :param debug: False/True or l2 to show level 2 debugging messages
+    """
+
+    def __init__(
+        self, inpath, outpath, columnName,
```

Comment:
The type hints for inpath, outpath and columnName are missing. I guess inpath and outpath should be `str | Path`

Reply:
(if you want to support Paths, I was never crazy about them)